src/main/java/com/google/api/codegen/py/PythonImportHandler.java

```diff
@@ -59,14 +61,15 @@ public PythonImportHandler(Interface service) {
     addImportExternal("google.gax", "path_template");
 
     // Add method request-type imports.
-    for (Method method : service.getMethods()) {
+    for (MethodConfig methodConfig : apiConfig.getInterfaceConfig(service).getMethodConfigs()) {
+      Method method = methodConfig.getMethod();
       addImport(
           method.getFile(),
           PythonImport.create(
               ImportType.APP,
               method.getFile().getProto().getPackage(),
-              PythonProtoElements.getPbFileName(method.getInputMessage())));
-      for (Field field : method.getInputType().getMessageType().getMessageFields()) {
+              PythonProtoElements.getPbFileName(method)));
```

Comment:
What is this change for?

Consider an API which defines all of its RPCs (e.g., `GetFoo`) in a file `foo_service.proto` and all of its request messages (e.g., `GetFooRequest`) in a separate file `foo_service_messages.proto`.

IIUC, previously we would import the file containing the request message type, `foo_service_messages_pb2`, but now we import the file containing the RPC definition, `foo_service_pb2`.

That seems wrong to me, since in the codegen we construct the request message type but don't actually reference the RPC definition.